pkgs/applications/audio/gst123/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, pkg-config
+, wrapGAppsHook
+, gst_all_1
+, gtk3
+, ncurses
+}:
+
+stdenv.mkDerivation rec {
```

Comment:
I think we should use the `finalAttrs` pattern here.

The `finalAttrs` pattern will let you remove the `rec` keyword ([see its implementation in Nix](https://github.com/NixOS/nix/blob/2d1d81114d72ace89ce08cd3bc93f4eb27a2975d/src/libexpr/nixexpr.cc#L121)).

Why is this pattern preferred to `rec` ?

Let's take this simple code example:

```nix
mkDerivation rec {
   foo = 1;
   bar = foo + 1;
}
```

and then `.overrideAttrs(old: { foo = 2; })`, you'll get `{ foo = 2; bar = 2; }` while with `finalAttrs` pattern, it would work correctly because it's a real fixed point.

Let me share a couple of useful links regarding the `finalAttrs` pattern:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Feel free to reach out if you need some assistance.

Reply:
I believe we should not overload newcomers with this `finalAttrs` pattern yet. We can just modernize the expression later.

Reply:
I did that with "newcomers" plenty of times and everything went well... why it would be an issue now?

Reply:
OK then. 
Now, instead of saying "use finalAttrs, it is nice and cute", show the places where the code should be modified.

Reply:
I've spent my time writing the explanation why this pattern should be used, with links. I tried to be complete so the new contributor is aware of what to do. Please, give them a chance to read and understand the reasoning behind this and come back if necessary. IMHO, I don't think that spotting each line where it should be changed is useful.

Reply:
Well, I am not really the right person to give a useful opinion on whether using `rec` or `finalAttrs` is better for new packages, but I changed it now.